Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -960,6 +969,10 @@ void WebProcessProxy::assumeReadAccessToBaseURL(WebPageProxy& page, const String
         weakPage->addPreviouslyVisitedPath(path);
         completionHandler();
     };
+
+    if (!networkProcessWillCheckBlobFileAccess())
```

Comment:
Nit: Should we NSLog something here to let the racy application author know they need to handle async calls properly?

Reply:
On second thought, since this is just a LOA check, it's not very useful to tell them they aren't getting the blob access check behavior.

Reply:
Yeah in this case it's pretty obvious.  They'll get callbacks in a different order, their application will break, and they'll fix it.